### PR TITLE
[enhancement](memory) Support jemalloc 64K page size - thirdparty

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Jemalloc compiled on a system with page size 4K can only run on a system with the same page size 4K.
If it is run on a system with page size > 4K, an error `unsupported system page size`.
Jemalloc compiled on a system with page size 64K can run on a system with page size < 64K, but this will waste more memory. Jemalloc does not support dynamic adaptation to the page size of the system.
The reason is that jemalloc will perform some optimizations based on the page size when compiling.

same as #28426

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -1066,13 +1066,15 @@ build_arrow() {
...
+    cp -rf ./jemalloc_ep-prefix/src/jemalloc_ep/dist/lib/libjemalloc_pic.a "${TP_INSTALL_DIR}/lib64/libjemalloc_arrow.a"
     cp -rf ./brotli_ep/src/brotli_ep-install/lib/libbrotlienc-static.a "${TP_INSTALL_DIR}/lib64/libbrotlienc.a"
     cp -rf ./brotli_ep/src/brotli_ep-install/lib/libbrotlidec-static.a "${TP_INSTALL_DIR}/lib64/libbrotlidec.a"
     cp -rf ./brotli_ep/src/brotli_ep-install/lib/libbrotlicommon-static.a "${TP_INSTALL_DIR}/lib64/libbrotlicommon.a"
     strip_lib libarrow.a
-    strip_lib libjemalloc.a
+    strip_lib libjemalloc_arrow.a
     strip_lib libparquet.a
+
+    cp -rf "${TP_INSTALL_DIR}/lib64/libjemalloc_arrow.a" "${TP_INSTALL_DIR}/lib64/libjemalloc.a" # TODO delete
```
Comment (adonis0147) [inline thirdparty/build-thirdparty.sh:1077]: This line will overwrite the static library (libjemalloc.a) built before in the function `build_jemalloc`. Is it right?

Comment (xinyiZzz) [reply thirdparty/build-thirdparty.sh:1077]: `build_jemalloc` set the suffix `--with-install-suffix="_doris"`, the compiled static library is `libjemalloc_doris.a`. 

Here `cp -rf libjemalloc_arrow.a libjemalloc.a` is just for compatibility, this line will be deleted later, only `libjemalloc_arrow.a` and `libjemalloc_doris.a` in `thirdparty/install/lib`

image

**Review by adonis0147 — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**